store.go

```diff
@@ -1051,15 +1052,20 @@ func (s *store) imageTopLayerForMapping(image *Image, ristore ROImageStore, crea
 			}
 		}
 		// Walk the top layer list.
+		found := false
 		for _, candidate := range append([]string{image.TopLayer}, image.MappedTopLayers...) {
 			if cLayer, err := store.Get(candidate); err == nil {
 				// We want the layer's parent, too, if it has one.
 				var cParentLayer *Layer
 				if cLayer.Parent != "" {
-					// Its parent should be around here, somewhere.
-					if cParentLayer, err = store.Get(cLayer.Parent); err != nil {
-						// Nope, couldn't find it.  We're not going to be able
-						// to diff this one properly.
+					// Its parent should be in one of the stores, somewhere.
+					for _, ps := range allStores {
+						if cParentLayer, err = ps.Get(cLayer.Parent); err == nil {
+							found = true
```

Comment:
you could use a label for the loop you want to address:
```
layers: for _, candidate := range append([]string{image.TopLayer}, image.MappedTopLayers...) {
[...]
       continue layers;
[...]
```

so you won't need the additional variable

Reply:
I don't want to breakout of the inner loop unless found.  Now I can check if cParentLayer != nil and then continue outside the loop.